### PR TITLE
Support non-ascii on JS.String.toLowerCase and toUpperCase

### PR DESCRIPTION
**Pull request by pedrobslisboa**
## Description

This PR adds the support of non-ascii on `JS.String.toLowerCase` and `toUpperCase`

## How

I am using the `uucp` library to support it: https://github.com/ocaml-community/Camomile/tree/main
And I don't know about performance with others way, but I ran the bench with this code:
```reason
[@react.component]
let make = () => {
  let _ = Js.String.toLowerCase("Hello World");
  <html>
    <body>
      <h1> {React.string("Hello World")} </h1>
      <p> {React.string("This is an example")} </p>
    </body>
  </html>;
};
```
This branch (uucp):
Captura de Tela 2024-10-25 às 10 21 02

Main branch (without uucp):
image

### CODE REVIEW

**Comment by pedrobslisboa** [comment]
Missed adding camomile to Dune, as soon as I back from lunch will add it.

File: demo/universal/native/lib/HelloWorld.re [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [@react.component]
 let make = () => {
+  let _ = Js.String.toLowerCase("Hello World");
```
[the remark this answers is not in the document]
Comment (davesnx) [reply demo/universal/native/lib/HelloWorld.re:3]: the opposite, make it part of the render!

**Comment by pedrobslisboa** [comment]
@davesnx I just added more tests and they expose an issue related to universal and native.

For the melange, we need to pass a js string, like `Js.String.toLowerCase {j|\u03a3|j}`
 https://melange.re/v4.0.0/playground/?language=OCaml&code=let+_+%3D+%28Js.String.toLowerCase+%22%CE%A3%22%29+%7C%3E+print_endline%0Alet+_+%3D+Js.String.toLowerCase+%7Bj%7C%5Cu03a3%7Cj%7D+%7C%3E+print_endline%0Alet+_+%3D+%22%5Cu%7B03a3%7D%22+%7C%3E+print_endline%0A&live=off

But for OCaml we need to have the unicode on ocaml form: `Js.String.toLowerCase "\u{03a3}"`

**Comment by anmonteiro** [comment]
did you consider using uutf instead of camomile? I think camomile is not really maintained + it doesn't support the latest unicode

**Comment by pedrobslisboa** [comment]
> did you consider using uutf instead of camomile? I think camomile is not really maintained + it doesn't support the latest unicode

I thought it had no support in handling to_lowercase or similar.
But I will take a look at it.

**Comment by anmonteiro** [comment]
you can use `uucp` for that https://erratique.ch/software/uucp/doc/Uucp/Case/index.html

**Comment by pedrobslisboa** [comment]
> you can use `uucp` for that https://erratique.ch/software/uucp/doc/Uucp/Case/index.html

@anmonteiro Thank you for sharing the tool; it not only works well but also performs better than Camomile.

**Pull request merged**